flutter_local_notifications/README.md

```diff
@@ -100,6 +101,38 @@ In the real world, this payload could represent the id of the item you want to d
 
 *Notes around initialisation*: if the app had been launched by tapping on a notification created by this plugin, calling `initialize` is what will trigger the `onSelectNotification` to trigger to handle the notification that the user tapped on. An alternative to handling the "launch notification" is to call the `getNotificationAppLaunchDetails` method that is available in the plugin. This could be used, for example, to change the home route of the app for deep-linking. Calling `initialize` will still cause the `onSelectNotification` callback to fire for the launch notification. It will be up to developers to ensure that they don't process the same notification twice (e.g. by storing and comparing the notification id).
 
+### [iOS only] Requesting notification permission
+
+By default this plugin will request notification permission when it is initialized. `IOSInitializationSettings` have three named parameters:
+1. `requestSoundPermission`,
+1. `requestBadgePermission`,
+1. `requestAlertPermission`
+that control this behaviour.
+
+If you want to delay permission request, set all of the above to false, then later call `requestPermissions` method with desired permissions.
```

Comment:
Hmm sorry I realised the intent you had went you had the "some logic" comment in the code further below now. How about breaking this sentence up. First would be "If you want to delay permission request, set all of the above to false..." that is followed by the code snippet for initialisation. After that is "then later call `requestPermissions` method with desired permissions" that then has the snippet for requesting permissions?